scipy/signal/signaltools.py

```diff
@@ -1020,8 +1020,9 @@ def convolve2d(in1, in2, mode='full', boundary='fill', fillvalue=0):
     val = _valfrommode(mode)
     bval = _bvalfromboundary(boundary)
 
-    with warnings.catch_warnings():
-        warnings.simplefilter('ignore', np.ComplexWarning)
+    with suppress_warnings() as sup:
```

Comment:
This just looks like a bit of a weird C-Code which creates first a double and then casts it to whatever is needed (sigtoolsmodule.c line 1052 is probably the reason). Not sure on first sight why this is done.